pipeline/objects/decorators.py

```diff
@@ -1,20 +1,17 @@
+from functools import partial, wraps
+
 from pipeline.objects.function import Function
 from pipeline.objects.graph_node import GraphNode
 from pipeline.objects.model import Model
 from pipeline.objects.pipeline import Pipeline
 from pipeline.objects.variable import Variable
 
 
-def pipeline_function(function):
-    """
-    Annotate a function as accesible by the Pipeline Object.
-
-        Parameters:
-                function (Function): user defined function.
-        Returns:
-                None.
-    """
+def pipeline_function(function=None, *, run_once=False, on_startup=False):
+    if function is None:
```

Comment:
maybe worth adding a comment explaining the reasoning behind this part (i.e. this is called when called with args, eg `@pipeline_function(run_once=True0`)

Reply:
Great idea, done